routes/users.js

```diff
@@ -0,0 +1,149 @@
+const router = require('express').Router();
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const validator = require("validator");
+const mongoose = require('mongoose');
+
+
+let User = require('../models/user');
+
+
+
+// Verify Token
+function verifyToken(req, res, next) {
+
+  // Get auth header value
+  const bearerHeader = req.headers['authorization']; 
+
+  // Check if bearer is undefined
+  if(typeof bearerHeader !== 'undefined'){
+
+  		const bearer = bearerHeader.split(' ');
+
+  		//Get token from array
+  		const bearerToken = bearer[1];
+
+  		//Set the Token
+  		req.token = bearerToken;
+
+  		//Next middleware
+  		next();
+  }//if
+  else
+  {
+  	//Forbidden
+  	res.sendStatus(403);
+  }//else
+
+  
+}//verifyToken
```

Comment:
this is way too much commenting. The code is self-explanatory.
Also in general not a fan of the commenting after } to say which block we were in. If we're using a code beautifier it should be easy enough to read and things should be aligned properly so we don't need them.